### DISCUSSION TITLE
issue: useForm does not provide correct types with Zod

### DISCUSSION THREAD

**Discussion by ryanpwaldon** [react-hook-form/react-hook-form#9681] [Q&A]
### Version Number

7.41.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-zod-ts-issue-jdz9uf?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

React hook form doesn't seem to be handling types correctly with Zod.

When I create a form with a zod schema, and try to access the property with `getValues`, or `watch`, etc – the inferred type is `string`, when the actual value is `undefined`.

Here's an example:

```tsx
const schema = z.object({ name: z.string() })

type Schema = z.infer<typeof schema>

const { getValues, watch, control } = useForm<Schema>({ resolver: zodResolver(schema) })

// Inferred type: string
const a = getValues('name') // undefined

// Inferred type: string
const b = watch('name') // undefined

// Inferred type: string
const c = useController({ control, name: 'name' }).field.value // undefined
```

Not sure whether this is intended behaviour or not?

### Expected behaviour

Inferred type should be `string | undefined`.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
>  etc – the inferred type is string, when the actual value is undefined.

This is due to missing `defaultValues` or `defaultValue`, the inferred type is correct and we may consider making `defaultValues` (next major) non-deep partial (required) to avoid these potential runtime errors.

**Comment by devklepacki** [comment]
I'm not really sure if that's connected, but there seems to be some issue with empty string handling with zod. https://stackoverflow.com/questions/73715295/react-hook-form-with-zod-resolver-optional-field/74932540

The thing is that with zod optional fields are still behaving as required.

There was also this issue: https://github.com/react-hook-form/react-hook-form/discussions/6980 and this one: https://github.com/react-hook-form/react-hook-form/issues/3985

They all seem connected with how an empty string's value is handled. If zod's `.preprocess()` is used, then the optional fields are handled correctly.

**Comment by bluebill1049** [comment]
thanks for the update @devklepacki yes it's related by default react hook form `defaultValues` are deep partial because it's optional to provide individual `defaultValue` for uncontrolled input and hence the conflict above.
